### PR TITLE
[fix][lmi] only use sequence iterators for generating outputs in stre…

### PR DESCRIPTION
**Pull request by siddvenk**
…aming cases

## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: engines/python/setup/djl_python/tests/test_rolling_batch.py [hunk 1]
```diff
@@ -28,9 +28,7 @@ def test_json_fmt(self):
         req2 = Request(req_input2)
         for req in [req1, req2]:
             req.set_next_token(Token(244, "He", -0.334532))
-            req.get_next_token()
```
[the remark this answers is not in the document]
Comment (siddvenk) [reply engines/python/setup/djl_python/tests/test_rolling_batch.py:31]: done!

File: engines/python/setup/djl_python/tests/test_rolling_batch.py [hunk 2]
```diff
@@ -398,10 +388,31 @@ def test_return_full_text(self):
...
+                "log_prob": -0.123123
+            }
+        }
+        req.reset_next_token()
         req.set_next_token(Token(4558, " world", -0.567854), True, 'length')
-        print(req.get_next_token(), end='')
-        assert json.loads(req.get_next_token().splitlines()[-1]) == {
+        next_token = req.get_next_token()
+        print(next_token, end='')
+        assert json.loads(next_token.splitlines()[-1]) == {
```
[the remark this answers is not in the document]
Comment (siddvenk) [reply engines/python/setup/djl_python/tests/test_rolling_batch.py]: done!

**Review by sindhuvahinis — APPROVED** [review]
